hasher.go

```diff
@@ -190,11 +192,8 @@ func (n *NmtHasher) HashLeaf(ndata []byte) ([]byte, error) {
 	minMaxNIDs = append(minMaxNIDs, nID...) // nID
 	minMaxNIDs = append(minMaxNIDs, nID...) // nID || nID
 
-	// add LeafPrefix to the ndata
-	leafPrefixedNData := make([]byte, 0, len(ndata)+1)
-	leafPrefixedNData = append(leafPrefixedNData, LeafPrefix)
-	leafPrefixedNData = append(leafPrefixedNData, ndata...)
-	h.Write(leafPrefixedNData)
+	h.Write([]byte{LeafPrefix}) //nolint:errcheck
+	h.Write(ndata)              //nolint:errcheck
```

Comment:
nit: there's a lint disable on line 180. Can we remove all the lint disables and actually check for errors?

Reply:
Done. TBH there is no sense to check 'cause hasher never returns any errors. It's even stated in `crypto.Hash` doc:

```
type Hash interface {
	// Write (via the embedded io.Writer interface) adds more data to the running hash.
	// It never returns an error.
	io.Writer
```

So we can drop this for sure (including line 180).